### PR TITLE
Fix time window effect on birth event selection range

### LINKED ISSUE #24: Missing entrires in results of query1-0.json
After executing the command with query1-0.json to obtain the specific users from the health dataset, COOL shows 8513 users. But the correct number of users should be 8592. The user ids that start with "P-19709" are all missed. The missed user ids are as follows.

```
 P-19709, P-19712, P-19713, P-19718, P-19720, P-19721, P-19732, P-19749, P-19762, P-19767,
 P-19772, P-19773, P-19776, P-19777, P-19778, P-19780, P-19782, P-19789, P-19790, P-19791,
 P-19797, P-19798, P-19799, P-19801, P-19803, P-19805, P-19806, P-19809, P-19812, P-19813,
 P-19820, P-19830, P-19834, P-19841, P-19842, P-19843, P-19844, P-19846, P-19860, P-19861,
 P-19864, P-19872, P-19874, P-19879, P-19882, P-19887, P-19890, P-19893, P-19901, P-19902,
 P-19907, P-19909, P-19910, P-19911, P-19915, P-19919, P-19923, P-19929, P-19931, P-19941,
 P-19944, P-19948, P-19950, P-19952, P-19954, P-19959, P-19962, P-19964, P-19972, P-19973,
 P-19975, P-19978, P-19983, P-19985, P-19989, P-19991, P-19992, P-19994, P-19998
```

### PR DESCRIPTION
**Pull request by hugy718**
Fix #24

### CODE REVIEW

**Comment by hugy718** [comment]
@KimballCai found another potential issue. In the same function. The birthOffset assignment on L485 does not match the comment. It is returned via cohort. Not sure if this is working as expected.
I tried switching the condition to (birthoffset > offset) It only affected the query result in one result tuple `ExtendedResultTuple(cohort=((1960, 1970]), age=0, measure=1594.0, min=0.0, max=85.0, sum=11888.0, num=381.0)` by increasing the measure by 1. Others are the same.

File: cool-core/src/main/java/com/nus/cool/core/cohort/ExtendedCohortSelection.java [hunk 1]
```diff
@@ -423,7 +423,7 @@ private int getUserBirthTime(int start, int end) {
 				// with time window
 				int startDay = firstDay;
 				int wlen = window.getLength();
-				int endDay = maxDate - (wlen - 1); // at least one time window
+				int endDay = TimeUtils.getDate(timeVector.get(end - 1)); // it is open for the time window to overflow to date not seen in dataset. We don't search beyong the last date user record appears
 				int windowOffset;
```
Comment (KimballCai) [inline cool-core/src/main/java/com/nus/cool/core/cohort/ExtendedCohortSelection.java:427]: we should filter users who do not have records for at least `wlen` days.

**Review by KimballCai — CHANGES_REQUESTED** [review]
need to add the logic to filter users by the time winsow setting.

**Comment by KimballCai** [comment]
Fix the issue #24 with the problem about the TimeWindow

**Pull request merged**